### DISCUSSION TITLE
Why can't I find the template file for NSI

### DISCUSSION THREAD

**Discussion by EHW-liao** [tauri-apps/tauri#7504] [Q&A] [answered]
I have set the template file for NSIS in tauriconf.json and placed it in the src-tauri directory, but during the build process, it always shows that the specified file cannot be found. The settings are as follows
“nsis”:{
        “template”:"oneInstall.nsi"
}

**Comment by EHW-liao** [comment]
[✔] Environment
    - OS: Windows 10.0.17134 X64
    ✔ WebView2: 115.0.1901.183
    ✔ MSVC: Visual Studio ���ɹ��� 2022
    ✔ rustc: 1.70.0 (90c541806 2023-05-31)
    ✔ Cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (environment override by RUSTUP_TOOLCHAIN)
    - node: 16.18.1
    - yarn: 1.22.19
    - npm: 8.19.2

[-] Packages
    - tauri [RUST]: 1.4.1
    - tauri-build [RUST]: 1.4.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.4.0
    - @tauri-apps/cli [NPM]: 1.4.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:3001/
    - framework: Vue.js
    - bundler: Vite

**Comment by FabianLars** [comment]
Where did you save the file? _Next_ to the tauri.conf.json file? Also, can you show the full error when building with the `--verbose` flag?

**Reply by EHW-liao** [reply]
yes, Next to the tauri.conf.json file ,  now the error is could not find: "FileAssociation.nsh"

       Info [tauri_bundler::bundle::windows::nsis] Target: x64
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("manufacturer")], "manufacturer")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("product_name")], "product_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("version")], "version")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("version_with_build")], "version_with_build")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("short_description")], "short_description")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("install_mode")], "install_mode")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("license")], "license")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("installer_icon")], "installer_icon")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("sidebar_image")], "sidebar_image")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("header_image")], "header_image")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("main_binary_name")], "main_binary_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("main_binary_path")], "main_binary_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("bundle_id")], "bundle_id")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("out_file")], "out_file")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("arch")], "arch")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("additional_plugins_path")], "additional_plugins
_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("allow_downgrades")], "allow_downgrades")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("display_language_selector")], "display_language
_selector")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("install_webview2_mode")], "install_webview2_mod
e")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("webview2_installer_args")], "webview2_installer
_args")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("webview2_bootstrapper_path")], "webview2_bootst
rapper_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("webview2_installer_path")], "webview2_installer
_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("copyright")], "copyright")))
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("languages"), valu
e: Context(Array [String("SimpChinese"), String("English")], ["languages"]) }], hash: {}
       Debug [handlebars::render] Rendering value: Path(Relative(([], "this")))
       Debug [handlebars::render] Rendering value: Path(Relative(([], "this")))
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("language_files"),
 value: Context(Array [String("SimpChinese.nsh"), String("English.nsh")], ["language_files"]) }], hash: {}
       Debug [handlebars::render] Rendering value: Path(Relative(([], "this")))
       Debug [handlebars::render] Rendering value: Path(Relative(([], "this")))
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("resources"), valu
e: Context(Object {}, ["resources"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("binaries"), value
: Context(Object {}, ["binaries"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("file_associations
"), value: Missing }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("resources"), valu
e: Context(Object {}, ["resources"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("binaries"), value
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("resources"), value: Con
text(Object {}, ["resources"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("binaries"), value: Cont
ext(Object {}, ["binaries"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("file_associations"), va
lue: Missing }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("resources"), value: Con
text(Object {}, ["resources"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("binaries"), value: Cont
ext(Object {}, ["binaries"]) }], hash: {}
       Debug [handlebars::render] Rendering helper: "each", params: [PathAndJson { relative_path: Some("file_associations"), value: Missing
 }], hash: {}
     Running [tauri_bundler::bundle::windows::nsis] makensis.exe to produce D:\Rust\tauriapp-0714\One-client\src-tauri\target\release\bundl
e\nsis\test_0.0.0_x64-setup.exe
     Running [tauri_bundler::bundle::common] Command `C:\Users\abchina\AppData\Local\tauri\NSIS\makensis.exe  -V3 D:\Rust\tauriapp-0714\One
-client\src-tauri\target\release\nsis\x64\installer.nsi`
MakeNSIS v3.08 - Copyright 1999-2021 Contributors
See the file COPYING for license details.
Credits can be found in the Users Manual.

Processing config: C:\Users\abchina\AppData\Local\tauri\NSIS\nsisconf.nsh
Processing script file: "D:\Rust\tauriapp-0714\One-client\src-tauri\target\release\nsis\x64\installer.nsi" (UTF16LE)
!include: could not find: "FileAssociation.nsh"
Error in script "D:\Rust\tauriapp-0714\One-client\src-tauri\target\release\nsis\x64\installer.nsi" on line 5 -- aborting creation process

**Reply by FabianLars** [reply, marked as answer]
I think that's a branch conflict, try the template from the 1.x branch: https://github.com/tauri-apps/tauri/blob/1.x/tooling/bundler/src/bundle/windows/templates/installer.nsi - i will update the link in the docs...

**Reply by EHW-liao** [reply]
it works , thanks bro
